floss/language/go/extract.py

```diff
@@ -0,0 +1,318 @@
+        .text:000000000048FFC7 66 81 78 0C 61 6E                             cmp     word ptr [rax+0Ch], 6E61h
+        .text:000000000048FFCD 75 49                                         jnz     short loc_490018
+        .text:000000000048FFCF 80 78 0E 6B                                   cmp     byte ptr [rax+0Eh], 6Bh ; 'k'
+        .text:000000000048FFD3 75 43                                         jnz     short loc_490018
+        """
+        extract_stackstring_pattern = re.compile(
+            b"\x48\xba(........)|\x48\xb8(........)|\x81\x78\x08(....)|\x81\x79\x08(....)|\x66\x81\x78\x0c(..)|\x66\x81\x79\x0c(..)|\x80\x78\x0e(.)|\x80\x79\x0e(.)"
+        )
+
+        # The "?=" in the regular expression is a lookahead assertion that allows us to match a specific pattern without including it in the actual match.
+        # The "re.DOTALL" flag ensures that the dot "." in the regular expression matches any character, including newline characters.
+
+    elif pe.FILE_HEADER.Machine == pefile.MACHINE_TYPE["IMAGE_FILE_MACHINE_I386"]:
+        arch = "386"
+
+        """
+        .text:0048CED3 75 6D                                         jnz     short loc_48CF42
+        .text:0048CED5 81 7D 00 50 61 73 73                          cmp     dword ptr [ebp+0], 73736150h
+        .text:0048CEDC 75 64                                         jnz     short loc_48CF42
+        .text:0048CEDE 66 81 7D 04 77 6F                             cmp     word ptr [ebp+4], 6F77h
+        .text:0048CEE4 75 5C                                         jnz     short loc_48CF42
+        .text:0048CEE6 80 7D 06 72                                   cmp     byte ptr [ebp+6], 72h ; 'r'
+        .text:0048CEEA 75 56                                         jnz     short loc_48CF42
+        """
+        extract_stackstring_pattern = re.compile(
+            b"\x81\xf9(....)|\x81\x38(....)|\x81\x7d\x00(....)|\x81\x3B(....)|\x66\x81\xf9(..)|\x66\x81\x7b\x04(..)|\x66\x81\x78\x04(..)|\x66\x81\x7d\x04(..)|\x80\x7b\x06(.)|\x80\x7d\x06(.)|\x80\xf8(.)|\x80\x78\x06(.)",
+            re.DOTALL,
+        )
+
+        # The "?=" in the regular expression is a lookahead assertion that allows us to match a specific pattern without including it in the actual match.
+        # The "re.DOTALL" flag ensures that the dot "." in the regular expression matches any character, including newline characters.
+    else:
+        raise ValueError("unhandled architecture")
+
+    for section in pe.sections:
+        try:
+            section_name = section.Name.partition(b"\x00")[0].decode("utf-8")
+        except UnicodeDecodeError:
+            continue
+
+        section_va = section.VirtualAddress
+        section_size = section.SizeOfRawData
+        section_data = section.get_data(section_va, section_size)
+
+        if section_name == ".text":
+            text_segment_data = section_data
+            text_segment_va = section_va
```

Comment:
I propose to call the `xrefs_in_*_segment` functions here directly and extend the xrefs list, otherwise there could be exceptions with missing section data